### PR TITLE
Fix MSButton storyboard crash

### LINKED ISSUE #8: Interface Builder crashes when components are used on Storyboard
Hi!

I'm adding the OfficeUIFabric pod as described on the README.md and trying to use the components included into this library, but the Xcode's Interface Builder designer crashes if I use a custom class for a button changing it to `MSButton`.

The minimal reproduction of the error is contained on this repository https://github.com/OrlSan/officeuifabric-sb-demo

Although the application is actually compiling and running the Interface Builder is displaying a blank canvas just containing the base `UIViewController` frames, but nothing else:

Captura de Pantalla 2019-05-22 a la(s) 16 25 04

Xcode is displaying the Issue message `Main.storyboard: error: IB Designables: Failed to render and update auto layout status for ViewController (BYZ-38-t0r): The agent crashed
`

The Diagnostic report found on `~/Library/Logs/DiagnosticReports` is pointing the error as the following:

```
Application Specific Information:
CoreSimulator 587.35 - Device: iPhone 8 (72039265-FC19-42F2-BCF0-99B04E1058F5) - Runtime: iOS 12.2 (16E226) - DeviceType: iPhone 8
/Users/orlando/Desktop/OfficeUIFabric Demo/Pods/OfficeUIFabric/OfficeUIFabric/Controls/MSButton.swift: 27: 12: Fatal error: Use of unimplemented initializer 'init(frame:)' for class 'OfficeUIFabric.MSButton'
```

The full crash log is here: [IBDesignablesAgent-iOS_2019-05-22-161443_d4nger.crash.txt](https://github.com/OfficeDev/ui-fabric-ios/files/3209427/IBDesignablesAgent-iOS_2019-05-22-161443_d4nger.crash.txt)

## What I've tried

As fas as I'm understanding the `MSButton` class (subclass of `UIButton`) is crashing the designer because the `init(frame:)` initializer is not implemented. I actually tried to change the source code and added the initializer so the `MSButton` implementation will include the following:

```swift
    @objc public init(style: MSButtonStyle = .secondaryOutline) {
        self.style = style
        super.init(frame: .zero)
        initialize()
    }

    // Added this initializer
    @objc public override init(frame: CGRect) {
        super.init(frame: frame)
        initialize()
    }

    public required init?(coder aDecoder: NSCoder) {
        super.init(coder: aDecoder)
        initialize()
    }
```

But again Xcode is displaying an empty canvas and the error is `Main.storyboard: error: IB Designables: Failed to render and update auto layout status for BNInitialViewController (BYZ-38-t0r): Failed to load designables from path (null)`

I don't know if the implementation of the initializer is the correct way to address this issue because the IB is not displaying the canvas anyway. Seems like an issue with the Autolayout resolution or something related.

## Additional information

Xcode: Version 10.2.1 (10E1001)
macOS version: macOS Mojave Version 10.14.5
Pod version: OfficeUIFabric (~> 0.2.0)

I appeciate your help on this Issue, and if you have any questions please let me know.

Thanks!

### PR DESCRIPTION
**Pull request by OrlSan**
Closes #8

### CODE REVIEW

**Comment by OrlSan** [comment]
Inside the new `convenience init(style:)` initializer there is also a call to the `update()` function. I included this because the buttons inside the demo will get an erratic size if the function is not executed again.

Because this is a double call to the `update function` I can left the same `init(frame:)` initializer as the currently present implementation in the master branch. I don't know what's best for the project on this subject.

**Comment by vladfilyakov** [comment]
The reason for second `update` is due to fact that property `didSet`s are not being called inside initializers - you have to defer property assignments for that:
```swift
    @objc public convenience init(style: MSButtonStyle = .secondaryOutline) {
        self.init(frame: .zero)
        defer {
            self.style = style
        }
    }
```
But I think the better solution would be to just leave `init(style:)` alone and don't change it to convenience initializer.

File: OfficeUIFabric/Controls/MSButton.swift [hunk 1]
```diff
@@ -57,17 +57,23 @@ open class MSButton: UIButton {
...
+
+        update()
     }
 
     public required init?(coder aDecoder: NSCoder) {
         super.init(coder: aDecoder)
         initialize()
     }
 
+    public override init(frame: CGRect) {
```
Comment (vladfilyakov) [inline OfficeUIFabric/Controls/MSButton.swift]: Please move it up above `init(coder:)` since this is more "important" initializer.

Comment (OrlSan) [reply OfficeUIFabric/Controls/MSButton.swift]: I just moved it. And also the `init(style:)` implementation was reversed to its original form.

**Comment by OrlSan** [comment]
I agree the `init(style:)` should not be a convenience init, so we better left the original implementation. Therefore this PR is just adding the `init(frame:)` implementation.

**Pull request merged**